frontend/cypress/integration/common/overview.ts

```diff
@@ -0,0 +1,135 @@
+import { And, Before, Given, Then } from "cypress-cucumber-preprocessor/steps";
+
+const url = "/console";
+
+Before(() => {
+    // Focing to not stop cypress on unexpected errors not related to the tests.
+    // There are some random failures due timeouts/loadtime/framework that throws some error in the browser.
+    // After reviewing the tests failures, those are unrelated to the app, so,
+    // it needs this event to not fail the CI action due some "slow" action or similar.
+    // This is something to review in future iterations when tests are solid, but I haven't found a better way to
+    // solve this issue.
+    cy.on("uncaught:exception", (err, runnable) => {
+        return false;
+    });
+});
+
+Given('user opens the overview page', () => {
+    // Forcing "Pause" to not cause unhandled promises from the browser when cypress is testing
+    cy.visit(url + '/overview?refresh=0');
+});
+
+And('user clicks in the {string} view', (view) => {
+    cy.get('button[data-display-mode="' + view + '"]').click();
+});
+
+And(`user filters {string} namespace`, (ns) => {
+    cy.get('select[aria-label="filter_select_type"]')
+        .select('Namespace')
+        .should('have.value', 'namespace_search');
+    cy.get('input[aria-label="filter_input_value"]')
+        .should('exist')
+        .type(ns)
+        .type('{enter}');
+});
+
+And(`user filters {string} health`, (health) => {
+    cy.get('select[aria-label="filter_select_type"]')
+        .select('Health')
+        .should('have.value', 'health');
+    cy.get('select[aria-label="filter_select_value"]')
+        .select(health);
+});
+
+And(`user selects Health for {string}`, (type) => {
+    let innerId = '';
+    switch (type) {
+        case 'Apps':
+            innerId = 'app';
+            break;
+        case 'Workloads':
+            innerId = 'workload';
+            break;
+        case 'Services':
+            innerId = 'service';
+            break;
+    }
+    cy.get('button[aria-labelledby^="overview-type"]')
+        .should('exist')
```

Comment:
I could be wrong but I think that `cy.get(...)` and most of the cypress API will block and timeout/fail if the button does not exist so you don't need to include the `.should('exist')` assertions.

Reply:
Yes, but the "should('exit')" makes the trick run faster, instead to trigger the timeout and the retry defaults.
I've browsed a couple of examples and I see this in a common way.

Reply:
[From the docs](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#When-To-Assert), the default behavior with timeout/retries is a major feature of the tool. It also looks like the assertions share the same timeout as the command: https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Example-2-Additional-Assertions. If we can save having to do an assertion for existence for every element in every test since cypress is already doing that, it will save time and make the tests simpler to read/write.

Reply:
Let me review if I can remove some of them.
I remember I found some cases that I needed for the chain.

Reply:
I've updated the PR. I removed some unnecessary assertions, perhaps I was updating the code and I didn't check that it was redundant.
Let me know how it looks there.

Reply:
LGTM!